source/extensions/filters/http/ext_proc/ext_proc.cc

```diff
@@ -1297,26 +1303,50 @@ void Filter::onReceiveMessage(std::unique_ptr<ProcessingResponse>&& r) {
   case ProcessingResponse::ResponseCase::kRequestHeaders:
     setDecoderDynamicMetadata(*response);
     processing_status = decoding_state_.handleHeadersResponse(response->request_headers());
+    if (on_processing_response_) {
+      on_processing_response_->afterProcessingRequestHeaders(*response, processing_status,
+                                                             decoder_callbacks_->streamInfo());
+    }
     break;
   case ProcessingResponse::ResponseCase::kResponseHeaders:
     setEncoderDynamicMetadata(*response);
     processing_status = encoding_state_.handleHeadersResponse(response->response_headers());
+    if (on_processing_response_) {
```

Comment:
These code indeed have a lot of duplicates, and there is a room to reduce it, but not a blocker though.